r/src/arrow_types.h

```diff
@@ -240,7 +241,17 @@ std::shared_ptr<arrow::Array> Array__from_vector(SEXP x, SEXP type);
 std::shared_ptr<arrow::RecordBatch> RecordBatch__from_arrays(SEXP, SEXP);
 std::shared_ptr<arrow::RecordBatch> RecordBatch__from_dataframe(Rcpp::DataFrame tbl);
 
+namespace ds = ::arrow::dataset;
+namespace fs = ::arrow::fs;
```

Comment:
Can you substitute `fs` in `r/src/filesystem.cpp` everywhere, like you did for `dataset.cpp`?